test/select.spec.js

```diff
@@ -636,6 +639,31 @@ describe('ui-select tests', function () {
     expect(el.find('.select2-search-choice-close').length).toEqual(1);
   });
 
+  it('should clear selection (with object as source)', function() {
```

Comment:
Maybe it is best I want to do some major refactoring to this repo, for example it needs to have seperate directives for tagging etc.